### PR TITLE
typings and docs for expressions.py

### PR DESCRIPTION
**Pull request by TheNeuralBit**
I can follow-up with suggestions for other files as I get through them. LMK if you think this is worthwhile (and correct)

### CODE REVIEW

File: sdks/python/apache_beam/dataframe/expressions.py [hunk 1]
```diff
@@ -24,60 +30,77 @@ class Session(object):
...
 
-  def proxy(self):
+  def proxy(self):  # type: (Expression) -> T
     return self._proxy
 
   def __hash__(self):
     return hash(self._id)
 
-  def __eq__(self):
+  def __eq__(self, other):
```
Comment (robertwb) [inline sdks/python/apache_beam/dataframe/expressions.py:69]: Oops...

File: sdks/python/apache_beam/dataframe/expressions.py [hunk 2]
```diff
@@ -24,60 +30,77 @@ class Session(object):
   def __init__(self, bindings={}):
     self._bindings = dict(bindings)
 
-  def evaluate(self, expr):
+  def evaluate(self, expr):  # type: (Session, Expression) -> Any
```
Comment (robertwb) [inline sdks/python/apache_beam/dataframe/expressions.py]: Here (and throughout) I don't think the convention is to declare the type of the self parameter for methods.

File: sdks/python/apache_beam/dataframe/expressions.py [hunk 3]
```diff
@@ -24,60 +30,77 @@ class Session(object):
...
-  def __init__(self, proxy):
+  def __init__(
+      self,  # type: PlaceholderExpression
+      proxy  # type: T
+  ):
     super(PlaceholderExpression, self).__init__('placeholder', proxy)
+    """Initialize a placeholder expression.
+
+    Args:
+      proxy: A proxy object with the type expected to be bound to this expression. Used for type checking at pipeline construction time.
```
Comment (robertwb) [inline sdks/python/apache_beam/dataframe/expressions.py]: Long line.

File: sdks/python/apache_beam/dataframe/expressions.py [hunk 4]
```diff
@@ -93,8 +116,18 @@ def preserves_partition_by_index(self):
...
+  def __init__(
+      self,  # type: ConstantExpression
+      value,  # type: T
+      proxy=None  # type: Optional[T]
+  ):
+    """Initialize a constant expression.
+
+    Args:
+      value: The constant value to be produced by this expression.
+      proxy: (Optional) a proxy object with same type as `value` to use for rapid type checking at pipeline construction time. If not provided, `value` will be used directly.
```
Comment (robertwb) [inline sdks/python/apache_beam/dataframe/expressions.py]: Long line.

**Review by robertwb — COMMENTED** [review]
The types and comments all look correct to me--just some style nits. Thanks!

File: sdks/python/apache_beam/dataframe/expressions.py (hunk 2 above)
Comment (TheNeuralBit) [reply sdks/python/apache_beam/dataframe/expressions.py]: Done

File: sdks/python/apache_beam/dataframe/expressions.py (hunk 1 above)
Comment (TheNeuralBit) [reply sdks/python/apache_beam/dataframe/expressions.py:69]: Haha yeah. Thanks to mypy for finding this one :)

File: sdks/python/apache_beam/dataframe/expressions.py (hunk 3 above)
Comment (TheNeuralBit) [reply sdks/python/apache_beam/dataframe/expressions.py]: Fixed (here and elsewhere). I tried to run yapf on this, but it seems to not care about long lines in docstrings? I just wrapped the lines manually for now.

Comment (robertwb) [reply sdks/python/apache_beam/dataframe/expressions.py]: Huh. Maybe it figures adding newlines in string literals could introduce semantic changes to the code.

**Pull request merged**